src/client-c++/cli_main.cpp

```diff
@@ -305,84 +306,89 @@ int main(int argc, char** argv)
 
     if(sIsSuccess)
     {
-        sSfServerV1.mPrivateKeyPath = sArgs.mPrivateKeyPath;
         sSfServerV1.mUrl            = sArgs.mUrl;
         sSfServerV1.mEpwdPath       = sArgs.mEpwdPath;
         sSfServerV1.mCurlDebug      = sArgs.mDebug;
         sSfServerV1.mVerbose        = sArgs.mVerbose;
+        sSfServerV1.mPasswordPtr    = NULL;
 
         sSfArgsV1.mProject          = sArgs.mProject;
         sSfArgsV1.mComment          = sArgs.mComment;
         sSfArgsV1.mExtraServerParms = sArgs.mExtraServerParms;
     }
 
-    if(sIsSuccess && !sArgs.mPasswordEnvVar.empty())
+    if(sIsSuccess &&
```

Comment:
>  sf_client will not use the ENV if the ssh-agent has been loaded with a private key.
Why is that an issue?  I suppose that would only be an issue if the environment is pointing to a different set of credentials than what they use loaded into ssh-agent?  Seems unlikely.  
> Another thing to consider would be to make an explicit ssh-agent flow on a --agent command.
This is buried in a lot of scripts including the crtSignedContainer.sh so they would all have to plumb through this option for something that normally is just handled seamlessly (and does with what I have here).  Also this isn't changing if the ssh-agent is used or not as that happens under the covers in the curl library so the option wouldn't actually be required unless you want to avoid the extra passwd prompt

Reply:
> Why is that an issue? I suppose that would only be an issue if the environment is pointing to a different set of credentials than what they use loaded into ssh-agent? Seems unlikely.

ssh-agent could have credentials to any number of systems other than the signing server. If the caller explicitly requests ENV, it sf_client should use it.

I'm not attached to -agent, but ENV flag should be respected.